karapace/avro_dataclasses/introspect.py

```diff
@@ -115,15 +115,11 @@ def _field_type(field: Field, type_: object) -> AvroType:  # pylint: disable=too
             raise UnderspecifiedAnnotation("Key and value types must be specified for map types")
         if args[0] is not str:
             raise UnsupportedAnnotation("Key type must be str")
-        return FieldSchema(
+        return MapType(
             {
                 "type": "map",
                 "values": _field_type(field, args[1]),
-                **(
-                    {"default": field.default_factory()}
-                    if field.default_factory is not MISSING
-                    else {}  # type: ignore[misc]
-                ),
+                **({"default": field.default_factory()} if field.default_factory is not MISSING else {}),
```

Comment:
Unfortunate victim to the cursed combination of 120 line-length limit + black ...